### PR TITLE
feat: Create toggle for skipping entity code creation

### PR DESCRIPTION
**Pull request by ywei390**
## Description

Work Item ID: [AB#41747](https://dev.azure.com/CluedIn-io/c054b4ae-1dab-43c2-af97-3683c744782f/_workitems/edit/41747)

## How has it been tested? 
Manually in local

### CODE REVIEW

File: src/ExternalSearch.Providers.Bregg/BrregExternalSearchProvider.cs [hunk 1]
```diff
@@ -240,20 +240,22 @@ public IEnumerable<Clue> BuildClues(ExecutionContext context, IExternalSearchQue
...
-            var code = GetOriginEntityCode(resultItem, request);
+            var clue = new Clue(request.EntityMetaData.OriginEntityCode, context.Organization);
+            PopulateMetadata(clue.Data.EntityData, resultItem, request, config);
 
-            var clue = new Clue(code, context.Organization);
-
-            PopulateMetadata(clue.Data.EntityData, resultItem, request);
-            clue.Data.EntityData.Codes.Add(request.EntityMetaData.OriginEntityCode);
+            var jobData = new BrregExternalSearchJobData(config);
+            if (!jobData.SkipEntityCodeCreation)
```
Comment (andrew-hardy) [inline src/ExternalSearch.Providers.Bregg/BrregExternalSearchProvider.cs:247]: I don't think this needs to be here as the re is similar code in PopulateMetadata

File: src/ExternalSearch.Providers.Bregg/BrregExternalSearchProvider.cs [hunk 2]
```diff
@@ -292,15 +294,19 @@ private CodeOrigin GetCodeOrigin()
             return CodeOrigin.CluedIn.CreateSpecific("brreg");
         }
 
-        public void PopulateMetadata(IEntityMetadata metadata, IExternalSearchQueryResult<BrregOrganization> resultItem, IExternalSearchRequest request)
+        public void PopulateMetadata(IEntityMetadata metadata, IExternalSearchQueryResult<BrregOrganization> resultItem, IExternalSearchRequest request, IDictionary<string, object> config)
         {
-            var code = GetOriginEntityCode(resultItem, request);
+            var jobData = new BrregExternalSearchJobData(config);
+            var code = jobData.SkipEntityCodeCreation ? request.EntityMetaData.OriginEntityCode : GetOriginEntityCode(resultItem, request);
```
Comment (andrew-hardy) [inline src/ExternalSearch.Providers.Bregg/BrregExternalSearchProvider.cs:300]: the code doesn't need to be conditional. I can always be request.EntityMetaData.OriginEntityCode

File: src/ExternalSearch.Providers.Bregg/BrregExternalSearchProvider.cs [hunk 3]
```diff
@@ -292,15 +294,19 @@ private CodeOrigin GetCodeOrigin()
...
 
             metadata.EntityType       = request.EntityMetaData.EntityType;
             metadata.Name             = request.EntityMetaData.Name;
             metadata.OriginEntityCode = code;
 
-            metadata.Codes.Add(code);
+            if (!jobData.SkipEntityCodeCreation)
+            {
+                metadata.Codes.Add(code);
```
Comment (andrew-hardy) [inline src/ExternalSearch.Providers.Bregg/BrregExternalSearchProvider.cs:308]: can be updated to `metadata.Codes.Add(GetOriginEntityCode(resultItem, request));`

**Review by andrew-hardy — CHANGES_REQUESTED** [review]

File: src/ExternalSearch.Providers.Bregg/BrregExternalSearchProvider.cs (hunk 1 above)
Comment (ywei390) [reply src/ExternalSearch.Providers.Bregg/BrregExternalSearchProvider.cs]: done

File: src/ExternalSearch.Providers.Bregg/BrregExternalSearchProvider.cs (hunk 2 above)
Comment (ywei390) [reply src/ExternalSearch.Providers.Bregg/BrregExternalSearchProvider.cs]: done update

File: src/ExternalSearch.Providers.Bregg/BrregExternalSearchProvider.cs (hunk 3 above)
Comment (ywei390) [reply src/ExternalSearch.Providers.Bregg/BrregExternalSearchProvider.cs]: done update

**Review by andrew-hardy — APPROVED** [review]

**Pull request merged**